### ISSUE TITLE
[Windows] Create a Visual Studio add-on for XAMLator

### ISSUE THREAD

**Issue by ylatuya** [ylatuya/XAMLator#16]
We are missing a Visual Studio add-on for XAMLator.

The vs4mac is less than 100 lines of code:
https://github.com/ylatuya/XAMLator/blob/master/XAMLator.Client.MonoDevelop/MonoDevelopIDE.cs
https://github.com/ylatuya/XAMLator/blob/master/XAMLator.Client.MonoDevelop/StartupHandler.cs

The add-on should implement the IIDE interface and start the monitor when the add-on is launched

**Comment by jsuarezruiz** [comment]
I will try to help with this!

**Comment by npagare** [comment]
Hi @jsuarezruiz, I am looking forward to test this. 

Is there a prerelease version you would like me to test it on my end ?

Thanks

**Comment by leo-mck** [comment]
@ylatuya @jsuarezruiz let me know if you need any help testing this ;)

**Comment by MagicAndre1981** [comment]
@jsuarezruiz are you still working on it? I can [only see 1 commit](https://github.com/jsuarezruiz/XAMLator/commit/4b9833c27cddf542d1ff0c4637408cbc37768b66)

**Comment by guiltm** [comment]
@jsuarezruiz are you working on it, or anyone? Thank you

**Comment by Sergiojoel9877** [comment]
Any updates?

**Comment by MagicAndre1981** [comment]
I switched to [Hotreload](https://github.com/AndreiMisiukevich/HotReload) which has a [VS extension](https://marketplace.visualstudio.com/items?itemName=StanislavBavtovich.hotreloadxamarinforms)

**Comment by ylatuya** [comment]
There is an addin for Visual Studio in this fork https://github.com/csteeg/XAMLator/, we need to get the original author to create a PR here :)

**Comment by jsuarezruiz** [comment]
We will have the Visual Studio for Windows the next week (PR). And after merge will be available in https://marketplace.visualstudio.com

**Comment by jsuarezruiz** [comment]
I would like to share some screenshots:

![XAMLator01](https://user-images.githubusercontent.com/6755973/54494495-2ce8c200-48db-11e9-9d26-70d8ba9f3a24.gif)
![XAMLator02](https://user-images.githubusercontent.com/6755973/54494500-340fd000-48db-11e9-8505-0304636e6667.gif)

**Comment by Sergiojoel9877** [comment]
Nice!

**Comment by npagare** [comment]
@jsuarezruiz thank you. I am going to download it in VS 15.9.9 and start using it.
Has it been published in https://marketplace.visualstudio.com ? I don't see it yet.

**Comment by ylatuya** [comment]
@jsuarezruiz  awesome work! I will review the PR tonight or tomorrow.
We also have to think about CI/CD and update the azure pipeline to do the Windows builds too.

**Comment by jsuarezruiz** [comment]
@IoTFier We need to merge the PR, pass some tests and integrate it in the CI/CD process. Will be available soon but is not available yet.

**Comment by guiltm** [comment]
Nice!

Em seg, 18 de mar de 2019 15:45, Javier Suárez Ruiz <
notifications@github.com> escreveu:

**Comment by npagare** [comment]
> @IoTFier We need to merge the PR, pass some tests and integrate it in the CI/CD process. Will be available soon but is not available yet.

@jsuarezruiz  - I am (and sure many of our community users) will be more than happy to be a beta tester if that helps. 

Thanks

**Comment by mzhukovs** [comment]
@jsuarezruiz Does this mean the hot reloading will work when running the UWP project (within a Xamarin Forms solution) as well?

**Comment by mzhukovs** [comment]
VS2019 now generally available, time to get that extension published!!! <3 <3

**Comment by IoTFier** [comment]
> VS2019 now generally available, time to get that extension published!!! <3 <3

I don't see it under VS Extensions yet. Is it in the marketplace ?

**Comment by toomasz** [comment]
Is that extension available already? Where can i donwnload it from?

**Comment by npagare** [comment]
> Is that extension available already? Where can i donwnload it from?

@toomasz - I don't think it is ready yet.
@jsuarezruiz - any ETA or input ?

**Comment by jsuarezruiz** [comment]
We updated the Windows extension two days ago: https://github.com/ylatuya/XAMLator/pull/56
We would like to launch it for VS 2017 and VS 2019 (available now). For that, we need more tests time for VS 2019, etc. If anyone want, I can share a package here (vsix).

**Comment by mzhukovs** [comment]
Yes plz share! @jsuarezruiz really looking forward to giving it a go. Seems like others are as well.

**Comment by toomasz** [comment]
@mzhukovs I think you can build extension yourself, it's located here: https://github.com/ylatuya/XAMLator/tree/master/XAMLator.Client.VisualStudio

**Comment by ylatuya** [comment]
The extension should also be available in the CI's artifacts

**Comment by mzhukovs** [comment]
I pulled down master and tried to run the VSIX after building but it says it couldn't find any currently installed compatible products. I only have VS2019 installed on the machine that I attempted this on. @toomasz

**Comment by npagare** [comment]
@ylatuya , @jsuarezruiz - Any hopes of getting this extension out for VS 2019 this week ?
Please help.

**Comment by npagare** [comment]
@ylatuya , @jsuarezruiz  - is this extn killed because Microsoft is trying to build a new one ?

**Comment by mzhukovs** [comment]
Yea curious what happened - seemed like you guys had so much momentum and got it very close, have you dropped plans to finish this piece? @ylatuya @jsuarezruiz

**Comment by npagare** [comment]
@ylatuya , @jsuarezruiz - is this extn killed because Microsoft is trying to build a new one ?
Are you planning to release this anytime soon ?

**Comment by toomasz** [comment]
@IoTFier You can try https://github.com/AndreiMisiukevich/HotReload - it has extension now

**Comment by npagare** [comment]
@toomasz - I am using that now, but there have been some instability with the latest changes (for good reasons) being delivered. Thank you.
